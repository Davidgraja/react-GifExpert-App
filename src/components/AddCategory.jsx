import { useState } from 'react';
import PropTypes from "prop-types";

export function AddCategory({onNewCategory , onChangeAmountGif}) {

    const [inputValue, setInputValue] = useState('');
    const [amout, setAmout] = useState("");

    const onInputChange = ({target}) =>{
        setInputValue(target.value);

    }


    const onSubmitEvent = (event) =>{
        event.preventDefault();
        const valueInput = inputValue.trim()
        const valueAmount = amout.trim()
        if(valueInput.length <=1) return;
        if(valueAmount <=0) return;
        
        onNewCategory(valueInput);
        onChangeAmountGif(valueAmount);
        setInputValue('');
        setAmout('');
        
    }

    const onAmoutChange = ({target}) => {
        setAmout(target.value)

    }

    

    return (
        <form onSubmit={ onSubmitEvent } aria-label="form" >

            <input 
                type="text" 
                placeholder='Buscar gifs'
                value={ inputValue }
                onChange={ onInputChange }
                data-testid = 'categoryInput'
                
            />

            <input 
                type="text" 
                pattern='^\d+$'
                title='Solo numeros'
                placeholder={'Cantidad'}
                value={ amout }
                onChange={ onAmoutChange }
                data-testid = 'amountInput'
            />
            
            <input type="submit" value={'Buscar'} className="btn-submit" />
            

        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired,
    onChangeAmountGif : PropTypes.func.isRequired
}
