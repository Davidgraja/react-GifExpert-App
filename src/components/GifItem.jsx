import PropsTypes from 'prop-types'

export const GifItem = ({title , url , id}) => {
    return (
        <div className="card">
            <img src={ url } alt={ title } />
            <p data-testid='p-title'>{ title }</p>
        </div>
    )
}

GifItem.propTypes = {
    title : PropsTypes.string.isRequired,
    url : PropsTypes.string.isRequired
}