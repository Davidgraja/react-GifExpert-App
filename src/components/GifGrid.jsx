import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({ category , onDeleteCategory }) => {
    const {images , isLoading} = useFetchGifs(category);

    const eventDelete = (e) =>{
        const category = e.nativeEvent.path[1].childNodes[0].innerText
        onDeleteCategory( category )
    }

    return (
        <>
            
            <section className="container-title-button">
                <h3>{ category }</h3>
                <button onClick={ eventDelete }>Eliminar</button>
            </section>
            
            

        {
            isLoading && (<h2>Cargando....</h2>)
        }

            
            <div className="card-grid">
                { images.map((image) => (

                    <GifItem 
                        key={ image.id }
                        {...image}
                    />

                ))}

            </div>
        </>
    )
}