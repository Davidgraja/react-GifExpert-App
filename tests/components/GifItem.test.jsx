import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"

describe('pruebas sobre el componente GifItem' , ()=>{

    const titleGif = "god of war";
    const urlGif = "https://image.api.playstation.com/vulcan/ap/rnd/202207/1210/4xJ8XB3bi888QTLZYdl7Oi0s.png";

    test('el compoente GifItem debe de recibir las propiedades title y url',()=>{
    
        render(<GifItem title={titleGif}  url={urlGif} />)
    })

    test('hacer la evaluacion frente el snapshot', ()=>{

        const {container} = render(<GifItem title={titleGif}  url={urlGif} />)
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el url y el alt indicado ', () => { 

        render(<GifItem title={titleGif}  url={urlGif} />)
        // expect(screen.getByRole('img').src).toBe(urlGif);
        const{src , alt} = screen.getByRole('img');
        expect(src).toBe(urlGif);
        expect(alt).toBe(titleGif);

    })

    test('debe de existir el titulo dentro del componente', () => { 
        render(<GifItem title={titleGif}  url={urlGif} />)
        expect(screen.getByText(titleGif)).toBeTruthy()
        // expect(screen.getByTestId('p-title').innerHTML).toBe(titleGif)
    })

})