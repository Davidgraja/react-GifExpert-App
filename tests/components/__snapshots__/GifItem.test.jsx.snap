// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el componente GifItem hacer la evaluacion frente el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="god of war"
      src="https://image.api.playstation.com/vulcan/ap/rnd/202207/1210/4xJ8XB3bi888QTLZYdl7Oi0s.png"
    />
    <p
      data-testid="p-title"
    >
      god of war
    </p>
  </div>
</div>
`;
